src/libcore/include/core/Window.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <functional>
+#include <memory>
+#include <string>
+
+namespace engine
+{
+namespace core
+{
+namespace detail
+{
+class WindowImpl;
+}
+
+class Window
```

Comment:
Pimpl idiom disallow stack allocation for window class. Policy pattern allows stack allocation, but does not hide implementation?
```suggestion
template<typename PlatformPolicy> class Window
```
Consider which one fits better to the Engine...